upgrade/upgrade-provider.go

```diff
@@ -331,6 +331,10 @@ func UpgradeProvider(ctx Context, repoOrg, repoName string) error {
 			"go", "get", "github.com/pulumi/pulumi-terraform-bridge/v3@"+targetBridgeVersion)).
 			In(repo.providerDir()))
 
+		steps = append(steps, step.Cmd(exec.CommandContext(ctx,
+			"go", "get", "github.com/hashicorp/terraform-plugin-framework")).
+			In(repo.providerDir()))
```

Comment:
What if the provider doesn't use (or compile against) the latest version of the bridge?

Reply:
Have we encountered such a situation, especially when upgrading the bridge explicitly? if so, then perhaps there's a guard flag somewhere we can use?

Reply:
I think we saw that for TLS. I'm generally nervous about adding to many guard flags. It makes the tool very hard to use.